bundlewrap/items/files.py

```diff
@@ -429,8 +430,31 @@ def test(self):
             ))
 
         if not self.attributes['delete'] and not self.attributes['content_type'] == 'any':
-            with self._write_local_file():
-                pass
+            with self._write_local_file() as local_path:
+                if self.attributes['test_with']:
+                    cmd = self.attributes['test_with'].format(quote(local_path))
+                    exitcode, stdout = self._run_validator(cmd)
+                    if exitcode == 0:
+                        io.debug("{i} passed local validation".format(i=self.id))
+                    elif exitcode in (126, 127, 255):
+                        io.debug("{i} failed local validation with code {c}, ignoring".format(i=self.id, c=exitcode))
```

Comment:
Thinking about it, it would be nice if we could show that message to the user. That's similar to `FaultUnavailable` while rendering a template, *we could not test the file because of an error*.